From https://bugzilla.redhat.com/attachment.cgi?id=436883

FIXME: See https://bugzilla.redhat.com/show_bug.cgi?id=620211, this
makes it build, but it may be still problematic at runtime.

diff -up ./c/mergepoints.c.python27 ./c/mergepoints.c
--- ./c/mergepoints.c.python27	2007-08-14 19:15:39.000000000 +0200
+++ ./c/mergepoints.c	2010-08-03 14:02:53.312329431 +0200
@@ -47,8 +47,8 @@ static char* NoControlFlowIfBuiltin[] = 
 
 /* instructions with a target: */
 #define HAS_JREL_INSTR(op)   (op == JUMP_FORWARD ||   \
-                              op == JUMP_IF_FALSE ||  \
-                              op == JUMP_IF_TRUE ||   \
+                              op == JUMP_IF_FALSE_OR_POP ||  \
+                              op == JUMP_IF_TRUE_OR_POP ||   \
                               op == FOR_ITER ||     \
                               /*    SETUP_LOOP replaced by FOR_ITER */    \
                               op == SETUP_EXCEPT ||   \
diff -up ./c/Python/pycompiler.c.python27 ./c/Python/pycompiler.c
--- ./c/Python/pycompiler.c.python27	2007-12-06 13:58:22.000000000 +0100
+++ ./c/Python/pycompiler.c	2010-08-03 14:03:10.274329433 +0200
@@ -2806,8 +2806,8 @@ code_t* psyco_pycompiler_mainloop(PsycoO
 		mp = psyco_next_merge_point(po->pr.merge_points, next_instr);
 		goto fine;
 
-	case JUMP_IF_TRUE:
-	case JUMP_IF_FALSE:
+	case JUMP_IF_TRUE_OR_POP:
+	case JUMP_IF_FALSE_OR_POP:
 		/* This code is very different from the original
 		   interpreter's, because we generally do not know the
 		   outcome of PyObject_IsTrue(). In the case of JUMP_IF_xxx
@@ -2816,7 +2816,7 @@ code_t* psyco_pycompiler_mainloop(PsycoO
 		cc = integer_NON_NULL(po, PsycoObject_IsTrue(po, TOP()));
 		if (cc == CC_ERROR)
 			break;
-		if (opcode == JUMP_IF_FALSE)
+		if (opcode == JUMP_IF_FALSE_OR_POP)
 			cc = INVERT_CC(cc);
 		if ((int)cc < CC_TOTAL) {
 			/* compile the beginning of the "if true" path */
